### PR TITLE
tutorials/physics/ray-casting: Update missing v4 fixes

### PR DESCRIPTION
**Pull request by karmaral**
There were a few missing method name changes (`_2d`, `_3d`, etc)

### CODE REVIEW

**Comment by karmaral** [comment]
I updated this because I was using it, but I realized there's a few more bits. The `intersect_ray` takes a params object, I'll adress that too and reopen it then.

**Comment by karmaral** [comment]
I did my best to update everything in a manner that reflects the rest of the information. Let me know if there's anything odd.

File: tutorials/physics/ray-casting.rst [hunk 1]
```diff
@@ -87,13 +87,13 @@ And in 3D:
...
     func _physics_process(delta):
-        var space_state = get_world().direct_space_state
+        var space_state = get_world_3d().direct_space_state
 
  .. code-tab:: csharp
 
     public override void _PhysicsProcess(float delta)
     {
-        var spaceState = GetWorld().DirectSpaceState;
+        var spaceState = GetWorld3d().DirectSpaceState;
```
Comment (Calinou) [inline tutorials/physics/ray-casting.rst:96]: The C# conversion is probably World3D (uppercase D), not World3d. cc @raulsntos

**Comment by TheYellowArchitect** [comment]
Very nice. While I haven't ran it locally, I finished some raycasting 3D code today and am confident what is written does work without problems.

**Review by mhilbrunner — APPROVED** [review]

**Comment by mhilbrunner** [comment]
Looks good to me, merged! Sorry this has been sitting for a while. Congrats on your first merged docs PR!

**Pull request merged**